### DISCUSSION TITLE
Check all dependencies of maven and the mostly used plugins based on security issues

### DISCUSSION THREAD

**Discussion by hendrikebbers** [support-and-care/maven-support-and-care#6] [Ideas for Funding]
We should do an initial check and define a best practice workflow (snyk GitHub Action, Dependabot, ...) to check the dependencies of a project. The defined workflow should be documented and implemented for the most important project in the maven ecosystem.

**Comment by Bukama** [comment]
I'm not exactly sure about the tools you have already posted but I know that Nexus IQ server can scan for licences and CVE.

Regardless of the goal to be free of CVEs there should be a classification what CVE scores are a no go and which are "we shall fix at some time, but for now it's okay" just to have that defined once before discussions start.
Not to mention things like "yeah this lib has this CVE, but we don't use it that we, so we are not affected". Should just be clarified and decision documentated.

**Comment by rmannibucau** [comment]
Why is it assumed it is not done?

**Reply by hendrikebbers** [reply]
I assume it is not done for all plugins and I have not found a doc that gives a guide for that for plugin developers.

**Reply by slawekjaranowski** [reply]
We need an example ... It will be easier to work on it.

In most of the cases resolving such issue is update or remove dependencies ....
What kind of guide you are looking here?

**Reply by rmannibucau** [reply]
doc are pointless cause not read whatever their quality, today we rely on release manager but setting up ossindex:audit can be an option even if I'm not convinced it will change anything security wise

**Comment by ascheman** [comment]
This discussion has been converted to Epic #66 for tracking.

Closing this discussion.
